File: BackEnd/backend/src/main/java/com/fitapp/backend/application/dto/subscription/response/SubscriptionResponse.java
package com.fitapp.backend.application.dto.subscription.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fitapp.backend.infrastructure.persistence.entity.enums.SubscriptionStatus;
import com.fitapp.backend.infrastructure.persistence.entity.enums.SubscriptionType;
import lombok.*;

import java.time.LocalDate;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SubscriptionResponse {
    @JsonProperty("id")
    private Long id;

    @JsonProperty("type")
    private SubscriptionType type;

    @JsonProperty("status")
    private SubscriptionStatus status;

    @JsonProperty("maxRoutines")
    private Integer maxRoutines;

    @JsonProperty("startDate")
    private LocalDate startDate;

    @JsonProperty("endDate")
    private LocalDate endDate;

    @JsonProperty("canExportRoutines")
    private boolean canExportRoutines;

    @JsonProperty("canAccessMarketplace")
    private boolean canAccessMarketplace;

    @JsonProperty("canSellOnMarketplace")
    private boolean canSellOnMarketplace;

    @JsonProperty("advancedAnalytics")
    private boolean advancedAnalytics;

    @JsonProperty("autoRenew")
    private boolean autoRenew;
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/response/subscription/response/SubscriptionResponse.kt
package com.fitapp.appfit.response.subscription.response

import com.google.gson.annotations.SerializedName

data class SubscriptionResponse(
    @SerializedName("id") val id: Long?,
    @SerializedName("type") val type: String,
    @SerializedName("status") val status: String?,
    @SerializedName("startDate") val startDate: String?,
    @SerializedName("endDate") val endDate: String?,
    @SerializedName("maxRoutines") val maxRoutines: Int?,
    @SerializedName("canExportRoutines") val canExportRoutines: Boolean = false,
    @SerializedName("canAccessMarketplace") val canAccessMarketplace: Boolean = false,
    @SerializedName("canSellOnMarketplace") val canSellOnMarketplace: Boolean = false,
    @SerializedName("advancedAnalytics") val advancedAnalytics: Boolean = false,
    @SerializedName("autoRenew") val autoRenew: Boolean = false
)